code/datums/actions/action.dm

```diff
@@ -28,12 +40,19 @@
 			IMG = image(target_obj.icon, button, target_obj.icon_state)
 		IMG.pixel_x = 0
 		IMG.pixel_y = 0
-		button.overlays += IMG
+		visual_references[VREF_IMAGE_LINKED_OBJ] = IMG
+		button.add_overlay(list(IMG))
```

Comment:
you dont need to wrap everything in a list before add and cut overlay they take non lists